### PR TITLE
[EXE-1564] Setup deploy to artifactory

### PR DESCRIPTION
**Pull request by dorisZ017**
Adding aiq version and scala version, deploying artifacts to jfrog

Moving over everything in `spark33` profile; upgrading `mssql-jdbc` to java 17

`mvn deploy`
```
Uploading to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/1.3.0-aiq0/spark-mssql-connector_2.12-1.3.0-aiq0.pom
Uploaded to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/1.3.0-aiq0/spark-mssql-connector_2.12-1.3.0-aiq0.pom (8.6 kB at 9.0 kB/s)
Uploading to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/1.3.0-aiq0/spark-mssql-connector_2.12-1.3.0-aiq0.jar
Uploading to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/1.3.0-aiq0/spark-mssql-connector_2.12-1.3.0-aiq0-sources.jar
Uploading to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/1.3.0-aiq0/spark-mssql-connector_2.12-1.3.0-aiq0-javadoc.jar
Uploaded to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/1.3.0-aiq0/spark-mssql-connector_2.12-1.3.0-aiq0-sources.jar (34 kB at 59 kB/s)
Uploaded to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/1.3.0-aiq0/spark-mssql-connector_2.12-1.3.0-aiq0.jar (81 kB at 130 kB/s)
Uploaded to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/1.3.0-aiq0/spark-mssql-connector_2.12-1.3.0-aiq0-javadoc.jar (37 kB at 55 kB/s)
Downloading from aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/maven-metadata.xml
Downloaded from aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/maven-metadata.xml (420 B at 2.4 kB/s)
Uploading to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/maven-metadata.xml
Uploaded to aiq-releases-artifactory: https://actioniq.jfrog.io/artifactory/aiq-sbt-local/com/microsoft/azure/spark-mssql-connector_2.12/maven-metadata.xml (367 B at 741 B/s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -187,6 +172,9 @@
                     <junitxml>.</junitxml>
                     <filereports>TestSuite.txt</filereports>
                     <stdout>W</stdout>
+                    <argLine>
+                        --add-opens=java.base/sun.nio.ch=ALL-UNNAMED
+                    </argLine>
```
Comment (dorisZ017) [inline pom.xml:177]: https://stackoverflow.com/questions/72724816/running-unit-tests-with-spark-3-3-0-on-java-17-fails-with-illegalaccesserror-cl

File: pom.xml [hunk 2]
```diff
@@ -220,7 +208,7 @@
                 <dependency>
                     <groupId>com.microsoft.sqlserver</groupId>
                     <artifactId>mssql-jdbc</artifactId>
-                    <version>8.4.1.jre8</version>
+                    <version>10.2.0.jre17</version>
```
Comment (dorisZ017) [inline pom.xml:211]: Spark didn't load `mssql-jdbc` jar to its classpath, supposedly due to incompatible java version. 10.2 is the earliest driver version that supports java 17: https://learn.microsoft.com/en-us/sql/connect/jdbc/microsoft-jdbc-driver-for-sql-server-support-matrix?view=sql-server-ver16#java-and-jdbc-specification-support

File: pom.xml [hunk 3]
```diff
@@ -19,6 +19,9 @@
     </licenses>
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <scala.binary.version>2.12</scala.binary.version>
```
Comment (MasterDDT) [inline pom.xml:22]: I guess you could have left this alone and made a new profile, `spark33-aiq` with all our version numbers? I guess this is fine too

File: pom.xml [hunk 4]
```diff
@@ -59,6 +62,17 @@
...
+        <dependency>
+            <groupId>org.scalatest</groupId>
+            <artifactId>scalatest_${scala.binary.version}</artifactId>
+            <version>3.2.6</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.microsoft.sqlserver</groupId>
+            <artifactId>mssql-jdbc</artifactId>
+            <version>10.2.0.jre17</version>
```
Comment (MasterDDT) [inline pom.xml:74]: Can you make this a variable and put it up in `properties`, just easier to read

File: pom.xml (hunk 3 above)
Comment (dorisZ017) [reply pom.xml:22]: I guess they have `spark33` in master branch just to distinguish among different spark versions? But this aiq branch is specifically for spark 3.3 so it's simpler to put everything here

File: pom.xml [hunk 5]
```diff
@@ -111,27 +125,12 @@
...
-                </executions>
-            </plugin>
-            
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.7.0</version>
                 <configuration>
-                    <source>1.8</source>
-                    <target>1.8</target>
+                    <source>17</source>
```
Comment (MasterDDT) [inline pom.xml:132]: This reminds me, I should have set this on our other forks hive/aws-glue/etc. We should make sure their tests pass on 17

**Comment by MasterDDT** [comment]
Did the tests also pass with `mvn deploy`?

File: pom.xml (hunk 2 above)
Comment (MasterDDT) [reply pom.xml]: Any changes to the connection string? Changed the param names?

**Review by MasterDDT — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (dorisZ017) [reply pom.xml]: Went through the release notes from 8.4 to 10.2, and the only change in connection string i `encrypt` and `trustServerCertificate`: https://learn.microsoft.com/en-us/sql/connect/jdbc/release-notes-for-the-jdbc-driver?view=sql-server-ver16#changes-in-102

**Comment by dorisZ017** [comment]
> Did the tests also pass with `mvn deploy`?

yes, both `mvn package` and `mvn deploy` run the tests

**Pull request merged**